### DISCUSSION TITLE
[FEATURE] Run server on device

### DISCUSSION THREAD

**Discussion by Immortalin** [libre-tube/LibreTube#467] [General]
**Describe the feature you'd like to see added**
Run server on device instead of relying on third party instances. The Piped backend is written in Java. The additional layer of redirection does not provide much benefit.

https://github.com/TeamPiped/Piped-Backend

**Additional context**
May want to consider calling NewPipeExtractor directly instead of relying on server.

**Comment by lilcheti** [comment]
Hmm the point of starting this app for me was that Youtube itself is banned in my country and I couldn't use newpipe or  yt vanced that easily. There is not much good documentation on using newpipe extractor but I think I can get some help from it developers.

**Comment by Wannacry-ops** [comment]
@Immortalin running the server on the device is faster? This is the benefit you say

**Comment by Bnyro** [comment]
That's out of scope for the project.
